src/scripts/main.js

```diff
@@ -1 +1,76 @@
 'use strict';
+
+function createMessage(text, className = 'success') {
+  const div = document.createElement('div');
+
+  div.classList = className;
+  div.setAttribute('data-qa', 'notification');
+  div.textContent = text;
+  document.body.append(div);
+}
+
+function addClickHandlers(mouseClick) {
+  document.addEventListener('click', mouseClick);
+  document.addEventListener('contextmenu', mouseClick);
+}
+
+function removeClickHandlers(mouseClick) {
+  document.removeEventListener('click', mouseClick);
+  document.removeEventListener('contextmenu', mouseClick);
+}
+
+const firstPromise = new Promise((resolve, reject) => {
+  const handleClick = (e) => {
+    if (e.button === 0) {
+      resolve('First promise was resolved');
+      removeClickHandlers(handleClick);
+    }
+  };
+
+  addClickHandlers(handleClick);
+
+  setTimeout(() => {
+    reject(new Error('First promise was rejected'));
+    removeClickHandlers(handleClick);
+  }, 3000);
+});
+
+const secondPromise = new Promise((resolve) => {
+  const handleClick = (e) => {
+    if (e.button === 0 || e.button === 2) {
+      resolve('Second promise was resolved');
+      removeClickHandlers(handleClick);
+    }
+  };
+
+  addClickHandlers(handleClick);
+});
+
+const thirdPromise = new Promise((resolve) => {
+  let leftClick = false;
+  let rightClick = false;
+
+  const handleClick = (e) => {
+    if (e.button === 2) {
+      e.preventDefault();
```

Comment:
The `e.preventDefault()` is used here to prevent the default context menu from appearing on right-click. Ensure this behavior is desired, as it might interfere with user expectations.